python/pyspark/daemon.py

```diff
@@ -100,7 +100,7 @@ def manager():
     def shutdown(code):
         signal.signal(SIGTERM, SIG_DFL)
         # Send SIGHUP to notify workers of shutdown
-        os.kill(0, SIGHUP)
+        os.kill(0, SIGTERM)
         sys.exit(code)
```

Comment:
`SIGHUP` is supposed to gracefully kill workers, and the we kill gracefully the daemon via proper `sys.exit`. Can you elabourate the process termination behaviours before and after this fix?

Reply:
Before the fix:
ps -ef|grep -i pyspark.daemon
/apps/anaconda3-5.3.0/bin/python -m pyspark.daemon
/apps/anaconda3-5.3.0/bin/python -m pyspark.daemon
/apps/anaconda3-5.3.0/bin/python -m pyspark.daemon

After the fix:
ps -ef|grep -i pyspark.daemon